File: flutter_app/lib/api/match_api.dart
import '../models/match_user.dart';
import 'api_client.dart';

class MatchApi {
  MatchApi(this._client);

  final ApiClient _client;

  Future<List<MatchUser>> recommended({int limit = 25}) async {
    final data = await _client.get('/match/recommended', query: {'limit': '$limit'}) as List<dynamic>;
    return data
        .map((dynamic item) => MatchUser.fromJson(Map<String, dynamic>.from(item as Map)))
        .toList();
  }
}

File: frontend/src/api/matchApi.ts
import { MatchUser, parseMatchUser } from '../models/matchUser';
import { asArray } from '../models/parse';
import { ApiClient } from './apiClient';

export class MatchApi {
  constructor(private readonly client: ApiClient) {}

  async recommended(limit = 25): Promise<MatchUser[]> {
    const data = asArray(
      await this.client.get('/match/recommended', {
        limit: String(limit),
      }),
    );
    return data.map(parseMatchUser);
  }
}
